src/sentry/api/endpoints/release_thresholds/release_threshold_details.py

```diff
@@ -11,14 +12,34 @@
 from sentry.api.exceptions import ResourceDoesNotExist
 from sentry.api.serializers import serialize
 from sentry.models.project import Project
+from sentry.models.release_threshold.constants import (
+    THRESHOLD_TYPE_STR_TO_INT,
+    TRIGGER_TYPE_STRING_TO_INT,
+    ReleaseThresholdType,
+)
+from sentry.models.release_threshold.constants import TriggerType as ReleaseThresholdTriggerType
 from sentry.models.release_threshold.release_threshold import ReleaseThreshold
 
 
+class ReleaseThresholdPUTSerializer(serializers.Serializer):
+    threshold_type = serializers.ChoiceField(choices=ReleaseThresholdType.as_str_choices())
+    trigger_type = serializers.ChoiceField(choices=ReleaseThresholdTriggerType.as_str_choices())
+    value = serializers.IntegerField()
+    window_in_seconds = serializers.IntegerField()
+
+    def validate_threshold_type(self, threshold_type: str):
+        return THRESHOLD_TYPE_STR_TO_INT[threshold_type]
+
+    def validate_trigger_type(self, threshold_type: str):
+        return TRIGGER_TYPE_STRING_TO_INT[threshold_type]
+
+
 @region_silo_endpoint
 class ReleaseThresholdDetailsEndpoint(ProjectEndpoint):
     owner: ApiOwner = ApiOwner.ENTERPRISE
     publish_status = {
         "GET": ApiPublishStatus.PRIVATE,
+        "PUT": ApiPublishStatus.PRIVATE,
```

Comment:
Do we plan on making these public once release thresholds are out?

Reply:
Yup!